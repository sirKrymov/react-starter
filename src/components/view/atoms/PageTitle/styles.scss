.page-title {
  &--mb {
    margin-bottom: 24px;
  }
}
