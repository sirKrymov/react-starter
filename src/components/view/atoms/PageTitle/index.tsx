// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

import cn from 'classnames';

import { Typography } from 'components/view/atoms/Typography';

import './styles.scss';

interface IProps {
  marginBottom?: boolean;
  className?: string;
  children: ReactNode;
}

export function PageTitle({
  marginBottom,
  className,
  children
}: IProps): ReactElement {
  const pageTitleClass = cn(
    {
      'page-title': true,
      'page-title--mb': marginBottom
    },
    className
  );

  return (
    <Typography className={pageTitleClass} variant="h1">
      {children}
    </Typography>
  );
}
