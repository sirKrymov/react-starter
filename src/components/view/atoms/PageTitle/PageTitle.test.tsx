import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { PageTitle } from '../PageTitle';

describe('<PageTitle> atom component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    shallowComponent = shallow(<PageTitle>Check test text</PageTitle>);
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
