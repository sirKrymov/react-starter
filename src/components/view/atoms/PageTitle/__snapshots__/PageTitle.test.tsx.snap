// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<Typography
  className="page-title"
  variant="h1"
>
  Check test text
</Typography>
`;
